prometheus/promhttp/instrument_client.go

```diff
@@ -49,21 +49,23 @@ func InstrumentRoundTripperInFlight(gauge prometheus.Gauge, next http.RoundTripp
 // http.RoundTripper to observe the request result with the provided CounterVec.
 // The CounterVec must have zero, one, or two non-const non-curried labels. For
 // those, the only allowed label names are "code" and "method". The function
-// panics otherwise. Partitioning of the CounterVec happens by HTTP status code
+// panics otherwise. For the "method" label a predefined default label value set
+// is used to filter given values. `additionalMethods` can be used to add more
+// methods to the set. Partitioning of the CounterVec happens by HTTP status code
 // and/or HTTP method if the respective instance label names are present in the
 // CounterVec. For unpartitioned counting, use a CounterVec with zero labels.
 //
 // If the wrapped RoundTripper panics or returns a non-nil error, the Counter
 // is not incremented.
 //
 // See the example for ExampleInstrumentRoundTripperDuration for example usage.
-func InstrumentRoundTripperCounter(counter *prometheus.CounterVec, next http.RoundTripper) RoundTripperFunc {
+func InstrumentRoundTripperCounter(counter *prometheus.CounterVec, next http.RoundTripper, additionalMethods ...string) RoundTripperFunc {
```

Comment:
Should we start the options pattern at this point? This would be safe for future options. Otherwise we would need major version bump of function deprecations.

I mean something like:

```
type Option func(*option)

WithExtraMethods([]string) Option
```
and
 
```suggestion
func InstrumentRoundTripperCounter(counter *prometheus.CounterVec, next http.RoundTripper, o ...Option) RoundTripperFunc {
```

?

Reply:
Even though current changes wouldn't also break the existing method calls, I like using options pattern. It would be more flexible. I'll update the PR.